flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/AbstractMetricGroup.java

```diff
@@ -116,15 +116,18 @@ public AbstractMetricGroup(MetricRegistry registry, String[] scope, A parent) {
 
 	@Override
 	public Map<String, String> getAllVariables() {
-		return internalGetAllVariables(0, Collections.emptySet());
+		return internalGetAllVariables(-1, Collections.emptySet());
```

Comment:
```suggestion
		return internalGetAllVariables(0, Collections.emptySet());
```
The PR. version causes an `ArrayIndexOutOfBoundsException` in [internalGetAllVariables:136](https://github.com/apache/flink/pull/13861/files#diff-cafa0bde2d752076fb6e568c8360a90d598b857c387a7263e4162fa37c06ca72R136).